src/components/Invoice/InvoiceForm.tsx

```diff
@@ -8,25 +8,26 @@ import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import { type SubmitHandler, useForm, Controller } from 'react-hook-form';
 import PersonalDataForm from '../PersonalDataForm';
-import { invoiceFormSchema, type IInvoiceForm } from '../Models/InvoiceForm.interface';
+import { invoiceFormSchema, type IInvoiceForm } from '../Models/Form/InvoiceForm.interface';
 import AmountForm from '../AmountsForm';
 import { zodResolver } from '@hookform/resolvers/zod';
-import { parse } from 'date-fns';
+import { format, parse } from 'date-fns';
+import { Link } from 'react-router-dom';
+import { useUpdateInvoice } from './use-update-invoice';
+import { useCreateInvoice } from './use-create-invoice';
 
 type InvoiceProps = {
   defaultValues: IInvoiceForm;
+  isNew?: boolean;
 };
 
 const parseDate = (dateString: string | null) => {
-  if (!dateString) {
-    return null;
-  }
-
-  return new Date(parse(dateString, 'yyyy-MM-dd', new Date()));
+  return dateString ? parse(dateString, 'yyyy-MM-dd', new Date()) : null;
 };
 
-export default function InvoiceForm({ defaultValues }: InvoiceProps) {
-  const onSubmit: SubmitHandler<IInvoiceForm> = (data) => console.log(data);
+export default function InvoiceForm({ defaultValues, isNew }: InvoiceProps) {
+  const { mutate } = !isNew ? useUpdateInvoice() : useCreateInvoice();
```

Comment:
I'd rather pass a function called `onSubmit` or something. That way you wouldn't need the `isNew` prop, and `useUpdateInvoice` and `useCreateInvoice` would be outside the form. Even if the boolean is stable and doesn't change over time, you are risking breaking hooks rules with yours composition (see more here: https://react.dev/warnings/invalid-hook-call-warning)

Reply:
done